runtime/dict.go

```diff
@@ -346,14 +393,43 @@ func (d *Dict) GetItem(f *Frame, key *Object) (*Object, *BaseException) {
 	if raised != nil {
 		return nil, raised
 	}
-	_, entry, raised := d.loadTable().lookupEntry(f, hash.Value(), key)
-	if raised != nil {
-		return nil, raised
+
+	var table dictTable
+	if rtable := d.loadReadTable(); rtable != nil {
+		table = *rtable
 	}
-	if entry != nil && entry != deletedEntry {
-		return entry.value, nil
+
+top:
+	if table != nil {
+		index, exists, raised := table.lookupEntry(f, hash.Value(), key)
+		if raised != nil || !exists {
+			return nil, raised
+		}
+		return table[index].value, nil
 	}
-	return nil, nil
+
+	d.mutex.Lock(f)
+	d.misses++
+	table = d.write
+	if table == nil {
+		table = *d.loadReadTable()
+		d.mutex.Unlock(f)
+		goto top
+	} else if d.misses > d.used {
```

Comment:
This line could use a comment. It's not clear to me what this comparison means.

Reply:
If more people is writing than reading, the _de facto_ table is the wrote one, not the read one, IIUC.